test/sdf_test.cpp

```diff
@@ -53,6 +61,162 @@ TEST(SDF, CubeVoid) {
+        }
+        return dvec;
+      },
+      {glm::vec3(-size), glm::vec3(size)}, edgeLength);
+
+  // EXPECT_EQ(levelSet.vertPos .size(), levelSetBatch.vertPos .size());
+
+  EXPECT_EQ(levelSet.triVerts.size(), levelSetBatch.triVerts.size());
+
+  Manifold gyroid(levelSet);
+  Manifold batchGyroid(levelSetBatch);
+
+  Box bounds = gyroid.BoundingBox();
+  Box boundsBatch = batchGyroid.BoundingBox();
+  const float precision = gyroid.Precision();
+  const float precisionBatch = batchGyroid.Precision();
+#ifdef MANIFOLD_EXPORT
+  if (options.exportModels) ExportMesh("batchcubeVoid.gltf", levelSetBatch, {});
+#endif
+
+  EXPECT_EQ(gyroid.Status(), Manifold::Error::NoError);
+  EXPECT_EQ(batchGyroid.Status(), Manifold::Error::NoError);
+  EXPECT_EQ(gyroid.Genus(), batchGyroid.Genus());
+  const float outerBound = size + edgeLength / 2;
+  EXPECT_NEAR(bounds.min.x, -outerBound, precision);
+  EXPECT_NEAR(bounds.min.y, -outerBound, precision);
+  EXPECT_NEAR(bounds.min.z, -outerBound, precision);
+  EXPECT_NEAR(bounds.max.x, outerBound, precision);
+  EXPECT_NEAR(bounds.max.y, outerBound, precision);
+  EXPECT_NEAR(bounds.max.z, outerBound, precision);
+  EXPECT_NEAR(boundsBatch.min.x, -outerBound, precisionBatch);
+  EXPECT_NEAR(boundsBatch.min.y, -outerBound, precisionBatch);
+  EXPECT_NEAR(boundsBatch.min.z, -outerBound, precisionBatch);
+  EXPECT_NEAR(boundsBatch.max.x, outerBound, precisionBatch);
+  EXPECT_NEAR(boundsBatch.max.y, outerBound, precisionBatch);
+  EXPECT_NEAR(boundsBatch.max.z, outerBound, precisionBatch);
+}
+
+TEST(SDF, BatchBounds) {
+  const float size = 4;
+  const float edgeLength = 0.5;
+
+  Mesh levelSet = LevelSetBatch(
+      [](std::vector<glm::vec3> pvec) {
+        std::vector<float> dvec(pvec.size());
+        for (int i = 0; i < pvec.size(); i++) {
+          dvec[i] = CubeVoid()(pvec[i]);
+        }
+        return dvec;
+      },
+      {glm::vec3(-size / 2), glm::vec3(size / 2)}, edgeLength);
+  Manifold cubeVoid(levelSet);
+  Box bounds = cubeVoid.BoundingBox();
+  const float precision = cubeVoid.Precision();
+#ifdef MANIFOLD_EXPORT
+  if (options.exportModels) ExportMesh("batchcubeVoid.gltf", levelSet, {});
+#endif
+
+  EXPECT_EQ(cubeVoid.Status(), Manifold::Error::NoError);
+  EXPECT_EQ(cubeVoid.Genus(), -1);
+  const float outerBound = size / 2 + edgeLength / 2;
+  EXPECT_NEAR(bounds.min.x, -outerBound, precision);
+  EXPECT_NEAR(bounds.min.y, -outerBound, precision);
+  EXPECT_NEAR(bounds.min.z, -outerBound, precision);
+  EXPECT_NEAR(bounds.max.x, outerBound, precision);
+  EXPECT_NEAR(bounds.max.y, outerBound, precision);
+  EXPECT_NEAR(bounds.max.z, outerBound, precision);
+}
+
+TEST(SDF, BatchSurface) {
+  const float size = 4;
+  const float edgeLength = 0.5;
+
+  Manifold cubeVoid(LevelSetBatch(
+      [](std::vector<glm::vec3> pvec) {
+        std::vector<float> dvec(pvec.size());
+        for (int i = 0; i < pvec.size(); i++) {
+          dvec[i] = CubeVoid()(pvec[i]);
```

Comment:
Are we basically duplicating existing SDF tests here? Perhaps we should make a helper to automatically run both types and compare them.

Reply:
I added a for-loop to automatically run the same checks on each function...